koa-unless/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for koa-unless 1.0
+// Project: https://github.com/Foxandxss/koa-unless
+// Definitions by: Bruno Krebs <https://github.com/brunokrebs/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import Koa = require("koa");
+
+export = unless;
+
+declare function unless(options: unless.Options | Function): unless.unlessWrapper;
+
+declare namespace unless {
+    export type unlessWrapper = (options: unless.Options | Function) => Koa.Middleware;
+
+    export interface Options {
+        custom?: () => boolean;
+        useOriginalUrl?: boolean;
+        path?: string | string[];
+        ext?: string | string[];
+        method?: string | string[];
+    }
+    export interface Middleware {
```

Comment:
Not sure if I understand how do you advise to define it. Something like this?

```typescript
export type Middleware = Koa.Middleware & { unless?: unlessWrapper };
```

And, to answer your question, the idea here is to be able to use this interface/type outside, like this:

```typescript
import koa = require('koa');
import unless = require('koa-unless');

const nonPngMiddleware: unless.Middleware = function *() {
    // original middleware that will be guarded by unless wrapper
};

nonPngMiddleware.unless = unless;

const app = new koa();
// uses 'nonPngMiddleware' only if the url requested does not ends with 'png'
app.use(nonPngMiddleware.unless({ ext: 'png' });

app.listen(3000);
```